packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -265,6 +265,36 @@ export function useBlockWrapperProps( props = {}, { __unstableIsHtml } = {} ) {
 		};
 	}, [ isNavigationMode, isHovered, setHovered ] );
 
+	useEffect( () => {
+		function getHoveredBlocksFromCursor( event ) {
+			const hoveredElements = document.elementsFromPoint(
+				event.clientX,
+				event.clientY
+			);
+			const blockIds = [];
+			hoveredElements.forEach( ( element ) => {
+				if ( element.dataset.block ) {
+					blockIds.push( element.dataset.block );
+				}
+			} );
+			return blockIds;
+		}
+		function onMouseEnter( event ) {
+			setHoveredBlocks( getHoveredBlocksFromCursor( event ) );
+		}
+		function onMouseLeave( event ) {
+			setHoveredBlocks( getHoveredBlocksFromCursor( event ) );
+		}
```

Comment:
This is our way of working around the inserter disruption (https://github.com/WordPress/gutenberg/pull/25348/files#r492402219).  Whenever a hover (enter/leave) fires, we parse the blocks under the cursor to retain the fidelity of the hovered blocks list.  

I don't currently see this adding performance issues, the worst case would be having a lot of nested blocks.  However, setting up ~22 nested blocks at 6x processor slowdown already has poor performance on master and running this PR I don't notice any observable impact to that.

Something I am uncertain about using this approach is how it will behave when we try iframe-ing editor content, depending on how that is set up and if this is trying to parse through nested iframes. 🤔

Reply:
> However, setting up ~22 nested blocks at 6x processor slowdown already has poor performance on master and running this PR I don't notice any observable impact to that.

I can second this. I played around with non-nested content with 15+ paragraph blocks, 10+ images, 10+ template parts, and, generally speaking, the performance seemed fine. 

With 4x and 6x slowdown, relative to the performance of the existing editor on master, there wasn't an observable impact on interaction latency. To be fair though, the editor is already pretty slow when on 4x and 6x slowdown.